### PR TITLE
Align make-bm-worker with dev-script's version

### PR DESCRIPTION
**Pull request by yprokule**
Commit https://github.com/openshift-metalkube/dev-scripts/commit/470a7bcbdc76db9ac7e9cf4086db79b418215fcf#diff-6aaa4e12b6fddc1b93b9ee994b50a49b introduced
ability to register both master and worker nodes.
This change brings in custom changes to make-bm-worker utility
to the BMO's repo.

### CODE REVIEW

File: cmd/make-bm-worker/main.go [hunk 1]
```diff
@@ -9,6 +9,11 @@ import (
 	"text/template"
 )
 
+const (
+	instanceImageSource      = "http://172.22.0.1/images/rhcos-ootpa-latest.qcow2"
+	instanceImageChecksumURL = instanceImageSource + ".md5sum"
```
Comment (dhellmann) [inline cmd/make-bm-worker/main.go]: The image related information is specific to the dev-scripts setup and using this code to deploy RHCOS nodes. We don't want to put that in this upstream repo.

File: cmd/make-bm-worker/main.go [hunk 2]
```diff
@@ -29,14 +34,35 @@ spec:
   bmc:
     address: {{ .BMCAddress }}
     credentialsName: {{ .Name }}-bmc-secret
+{{- if .MAC }}
+  bootMACAddress: {{ .MAC }}
+{{ end -}}{{- if .WithImage }}
+  userData:
+    namespace: openshift-machine-api
```
Comment (dhellmann) [inline cmd/make-bm-worker/main.go]: This setting is also specific to the environments we have downstream.

File: cmd/make-bm-worker/main.go [hunk 3]
```diff
@@ -29,14 +34,35 @@ spec:
   bmc:
     address: {{ .BMCAddress }}
     credentialsName: {{ .Name }}-bmc-secret
+{{- if .MAC }}
+  bootMACAddress: {{ .MAC }}
```
Comment (dhellmann) [inline cmd/make-bm-worker/main.go]: We shouldn't need MAC addresses for bare metal servers.

File: cmd/make-bm-worker/main.go [hunk 4]
```diff
@@ -29,14 +34,35 @@ spec:
...
+    namespace: openshift-machine-api
+    name: worker-user-data
+  image:
+    url: "{{ .ImageSourceURL }}"
+    checksum: "{{ .Checksum }}"
+{{ end -}}{{ if .WithMachine }}
+  machineRef:
+    name: {{ .Machine }}
+    namespace: {{ .MachineNamespace }}
+{{ end }}
```
Comment (dhellmann) [inline cmd/make-bm-worker/main.go]: The machine ref feature is OK.

File: cmd/make-bm-worker/main.go (hunk 1 above)
Comment (yprokule) [reply cmd/make-bm-worker/main.go]: Updated

File: cmd/make-bm-worker/main.go (hunk 2 above)
Comment (yprokule) [reply cmd/make-bm-worker/main.go]: Updated

**Pull request merged**